.github/workflows/tests.yml

```diff
@@ -253,8 +253,8 @@ jobs:
       - name: Run mypy
         run: mypy -p mesonpy
 
-  tests-pass:
```

Comment:
@dnicolodi moreover — this implementation was broken anyway.

https://github.com/marketplace/actions/alls-green#why

Reply:
It was not faulty. It just gets marked as skip instead of fail, but it's required to merge, so that's the same result. It would be better if it always ran, and reported the result (via alls-green) probably, but it still lets you do things like "merge when CI passes".

Why was this commented out? It doesn't slow the job by reducing parallism since it always runs at the end. So there's an extra at most 7 seconds or something tacked on the end. And it lets you makes the "passing" requirement in GitHub much simpler, and changes to it don't cause previous jobs to turn orange.

Reply:
> but it still lets you do things like "merge when CI passes".

Not exactly. It also let's you do things "merge when CI fails" because skipped==pass in branch protection.

Reply:
> It was not faulty.

Read the whole page at https://github.com/marketplace/actions/alls-green#why for explanation.

Reply:
Ahah, so that action is removing the "merge when CI fails" feature. :+1:

Reply:
> Why was this commented out? It doesn't slow the job by reducing parallism since it always runs at the end. So there's an extra at most 7 seconds or something tacked on the end. And it lets you makes the "passing" requirement in GitHub much simpler, and changes to it don't cause previous jobs to turn orange.

I commented it out because it is not used. Because all the projects within the mesonbuild organization share the GitHub Actions runners quota, it happened several times that the first batch of jobs from the meson-python is queued then some long-running jobs from the meson or wrapdb projects are queued and the summary job is left waiting for a long time for a runner just to execute an `echo`.

> And it lets you makes the "passing" requirement in GitHub much simpler

In general I don't think that having something enforcing the policy that a PR is to be merged only when the test suite passes is very useful. I think we can trust our pairs to behave. The test suite was failing on several configurations and these tests were simply excluded from the required set. The fact that merging was allowed made no one paid attentions to what was actually a bug. Just merge when all the tests are green is a much simpler and much more effective policy.

> changes to it don't cause previous jobs to turn orange

I don't know what you mean by this. I don't think the status of open PRs changes when the CI jobs definitions on the main branch change. If it would, that would be a welcome feature: PRs should be tested against the current main branch.